app/code/Magento/InventoryApi/Test/Api/StockSourceLink/AssignSourcesToStockTest.php

```diff
@@ -135,12 +135,28 @@ public function dataProviderWrongParameters(): array
             ],
             'nonexistent_source_id' => [
                 [-1, 20],
-                1,
+                10,
                 [
                     'rest_message' => 'Could not assign Sources to Stock',
                     'soap_message' => 'Could not assign Sources to Stock',
                 ],
             ],
+            'multiple_sources_assigned_to_default_stock' => [
```

Comment:
Need to move this check to `InventoryCatalog`
Yes, it will some code duplication but we need to do this for better modularity